cmd/kops-controller/main.go

```diff
@@ -81,6 +85,29 @@ func main() {
 	}
 
 	ctrl.SetLogger(klogr.New())
+	if opt.Server != nil {
+		var verifier fi.Verifier
+		var err error
+		switch opt.Server.Provider {
+		case kops.CloudProviderAWS:
+			verifier, err = awsup.NewAWSVerifier()
+			if err != nil {
+				setupLog.Error(err, "unable to create verifier")
+				os.Exit(1)
+			}
+		}
+
+		srv, err := server.NewServer(&opt, verifier)
+		if err != nil {
+			setupLog.Error(err, "unable to create server")
+			os.Exit(1)
```

Comment:
I've been finding recently that I've been rewriting the kubebuilder main to call into a `func run() error` function immediately, just to avoid having to log & exit every time!

Reply:
The structured logging, such as in the handling of `addNodeController`, makes it tricky to use that pattern.